src/Text/Pandoc/Writers/OpenDocument.hs

```diff
@@ -635,6 +650,46 @@ inlineToOpenDocument o ils
         addNote nn
         return nn
 
+mkLink :: WriterOptions -> [(Text,ReferenceType)] -> Text -> Text -> Doc Text -> Doc Text
+mkLink o identTypes s t d =
+  let findIdentType ident = mapMaybe identToType identTypes where
+        identToType (it,tp)
+          | it == ident = Just (it,tp)
+          | otherwise   = Nothing
+      maybeIdentAndType = case T.uncons s of
+                            Just ('#', ident) -> listToMaybe (findIdentType ident)
```

Comment:
The Prelude comes with a [`lookup`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:lookup) to handle associated lists, but only returns the second element of the tuple. If we want both we'd have to reconstruct the tuple.

```haskell
-- would be shorter with language extension TupleSections enabled, but let's not go there.
(\x -> (ident, x)) <$> lookup ident identTypes 
```

Possibly simpler would be to cast some pointfree [Data.Foldable.find](https://hackage.haskell.org/package/base-4.14.0.0/docs/Data-Foldable.html#v:find) magic:

```haskell
find ((ident ==) . fst) identTypes
```

The nice thing about the latter solution is that it works as long as `identTypes` is an instance of class `Foldable`. We wouldn't even have to touch such a line if we were to change `identTypes` to type `Map Text ReferenceType` later on.

Reply:
If you wonder how anyone can remember all those functions: we don't. The trick is to think about the types you need and to [put them into Hoogle](https://hoogle.haskell.org/?hoogle=(a%20-%3E%20Bool)%20-%3E%20%5Ba%5D%20-%3E%20Maybe%20a).

Reply:
done. Much better - thank you.